scala/private/phases/phase_compile.bzl

```diff
@@ -390,3 +390,18 @@ def _interim_java_provider_for_java_compilation(scala_output):
         compile_jar = scala_output,
         neverlink = True,
     )
+
+def _select_scalac(ctx):
```

Comment:
Could this `scalac` selection be done on toolchain? That would keep implicit toolchain attributes and selection by version closer to each other and would simplify `phase_compile`.